cpp/arcticdb/version/version_utils.hpp

```diff
@@ -160,6 +160,23 @@ inline void check_is_version(const AtomKey &key) {
     util::check(key.type() == KeyType::VERSION, "Expected version key type but got {}", key);
 }
 
+inline void verify_stream_id(const StreamId &stream_id) {
+    if (ConfigsMap::instance()->get_int("VersionStore.NoStrictSymbolCheck"))
+    {
+        ARCTICDB_DEBUG(log::version(), "Key with stream id {} will not be strictly checked because VersionStore.NoStrictSymbolCheck variable is set to 1.", stream_id);
+        return;
+    }
+
+    for (unsigned char c : std::get<std::string>(stream_id)) {
```

Comment:
Won't this blow up when the `stream_id` variant is a `NumericId`? I'm not sure how a `NumericId` could get passed in from the Python layer, but since the code has 

```
using NumericId = int64_t;
using StringId = std::string;
using VariantId = std::variant<NumericId, StringId>;
using StreamId = VariantId;
```
we should probably treat `StreamId` as a non-trivial variant everywhere.